src/utils/ExtrusionJunction.cpp

```diff
@@ -20,4 +20,17 @@ ExtrusionJunction::ExtrusionJunction(const Point p, const coord_t w, const coord
       region_id(region_id)
 {}
 
+void ExtrusionJunction::Swap(ExtrusionJunction* other) {
+  if (other == this) return;
+  InternalSwap(other);
+}
+void ExtrusionJunction::InternalSwap(ExtrusionJunction* other) {
+  using std::swap;
+
+  swap(p, other->p);
+  swap(w, other->w);
+  swap(perimeter_index, other->perimeter_index);
+  swap(region_id, other->region_id);
```

Comment:
I would make it absolutely clear this is the `std::swap` which is used for all four function call, certainly because due to our naming conventions (function: lowerCamelCase) we now have an actual class function called `swap(...)`
```suggestion
  std::swap(p, other->p);
  std::swap(w, other->w);
  std::swap(perimeter_index, other->perimeter_index);
  std::swap(region_id, other->region_id);
```

Reply:
While not in our code style standards, I agree that this is better practice.
Code was removed entirely, so this change is no longer necessary.